core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -451,7 +473,6 @@ public void setChannel(@Nonnull ChannelBuilder cb,
         if (headerStream == null) {
             LOGGER.log(Level.WARNING, "No header stream defined when setting channel for computer {0}. " +
                     "Launch log won't be printed", this);
-            headerStream = new NullOutputStream();
```

Comment:
This fix is brilliant, right?

Reply:
Will this not throw an NPE from the `PrintStream` constructor?

Reply:
Javadoc says it's acceptable. Will double-check the code

Reply:
Yes, we can just keep null here